x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/ResultSetTestCase.java

```diff
@@ -1748,4 +1753,8 @@ private Connection esWithLeniency(boolean multiValueLeniency) throws SQLExceptio
     private String asDateString(long millis) {
         return of(millis, timeZoneId);
     }
+
+    private ZoneId getZoneFromOffset(Long randomLongDate) {
```

Comment:
I think this reveals an issue with how we send the respond from the server.
Shouldn't we keep for example `Australia/Tasmania` instead of ` `+10:00`?
@costin @astefan what do you think?

Reply:
`+10:00` is static for any timestamp, but `Australia/Tasmania` offset might change (cause of test failure before).

Reply:
ZoneId is better I would argue though at the same time, if this is a cause for failure it might mean that the offset rules on the server are different than that on the client which is an issue on itself.

Reply:
I kind of agree that `+10:00` is ambiguous, but can there be an actual issue here?
Different offset rules on server and on the client can probably happen when different Java versions are used (and offset/timezone impl. differ between the two).

Reply:
The "issue" is that `+10:00` is static, it doesn't include DST and it doesn't include any changes in the offset of a timezone (in our example: `Australia/Tasmania`) that can happen over the years for certain timezones. So if the client would further need to process the result "looses" this info from the result returned, but it still poccesses the `Australia/Tasmania` since this was the timezone string it set as connection/REST property.